hail/src/main/scala/is/hail/expr/ir/Emit.scala

```diff
@@ -2415,393 +2438,395 @@ class Emit[C](
       PCode(pt, value))
   }
 
-  def deforestNDArray(x: IR, mb: EmitMethodBuilder[C], region: Value[Region], env: E): NDArrayEmitter[C] = {
+  def deforestNDArray(x0: IR, mb: EmitMethodBuilder[C], region: StagedRegion, env: E): EmitCode = {
     def emit(ir: IR, env: E = env): EmitCode =
       this.emitWithRegion(ir, mb, region, env, None)
 
-    def deforest(nd: IR): NDArrayEmitter[C] = deforestNDArray(nd, mb, region, env)
```

Comment:
why this change?

Reply:
Does my last comment (https://github.com/hail-is/hail/pull/9106#issuecomment-665034873) answer this? If not I can go into more detail.

Reply:
oh, sorry, I totally missed that!